src/script/common/helper.h

```diff
@@ -38,6 +39,12 @@ class LuaHelper
 	{
 		return lua_isnoneornil(L, index) ? default_value : readParam<T>(L, index);
 	}
+
+	/// Like lua_tonumber, but is guaranteed to handle nan, inf and -inf properly.
+	static double my_lua_string_to_double(lua_State *L, std::string_view sv);
+
+	/// Convert a double to string, handling nan, inf and -inf properly.
+	static std::string my_lua_double_to_string(double number);
```

Comment:
I think putting this in `LuaHelper` is a bit strange, since it's currently a clear single purpose class.